lib/Ohmbrewer_Relay.cpp

```diff
@@ -1,14 +1,33 @@
 #include "Ohmbrewer_Relay.h"
 #include "Ohmbrewer_Screen.h"
+#include "Ohmbrewer_Publisher.h"
+
 
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Equipment
- * @param powerPin - The power pin - on/off line. Digital pin number X.
+ * @param powerPin - The power pin - on/off line. Digital pin number X.TODO update
  * @param controlPin - The Control pin - Data/speed/power level Digital pin number X.
- */
-Ohmbrewer::Relay::Relay(int id, int powerPin, int controlPin) : Ohmbrewer::Equipment(id) {
-    _powerPin = powerPin;
+ * @param relayPins - controlPin always first in <list>
+ */
+Ohmbrewer::Relay::Relay(int id, std::list<int>* relayPins) : Ohmbrewer::Equipment(id) {
+    //FIXME
+    int size = relayPins->size();
+    int powerPin = -69;
+    int controlPin;
+    if  (size > 1){ // switch and relay
+        controlPin = relayPins->front();
+        relayPins->pop_front();
+        powerPin = relayPins->front();
```

Comment:
Given that the size is 2 (or at least you can expand that if to verify as much), you can use back() and not have to do any popping. There is a small, unlikely chance that whoever passed these relayPins in might want to continue using the list.